### ISSUE TITLE
ERROR : Assertion failed: cond.is_weights() && cond.weights().count() == 1 && "If condition must be a initializer!"

### ISSUE THREAD

**Issue by wwdok** [NVIDIA/TensorRT#938]
I want to convert TSN (Time-Sensitive Networking) model ([google drive file](https://drive.google.com/file/d/13rAilblasG3KZx8pXz1kxqkbL4Bmha59/view?usp=sharing)) to tensorrt, does it support now ? or it support other action recognition model ?

**Comment by wwdok** [comment]
I have just tried using trtexec to convert it to tensorrt file, but it reports error:
```
(base) weidawang@weidawang-TUF-Gaming-FX506LU-FX506LU:~/app/TensorRT-7.2.1.6/bin$ ./trtexec --onnx=tmp.onnx --saveEngine=tmp.engine
&&&& RUNNING TensorRT.trtexec # ./trtexec --onnx=tmp.onnx --saveEngine=tmp.engine
[12/03/2020-15:26:48] [I] === Model Options ===
[12/03/2020-15:26:48] [I] Format: ONNX
[12/03/2020-15:26:48] [I] Model: tmp.onnx
[12/03/2020-15:26:48] [I] Output:
[12/03/2020-15:26:48] [I] === Build Options ===
[12/03/2020-15:26:48] [I] Max batch: explicit
[12/03/2020-15:26:48] [I] Workspace: 16 MiB
[12/03/2020-15:26:48] [I] minTiming: 1
[12/03/2020-15:26:48] [I] avgTiming: 8
[12/03/2020-15:26:48] [I] Precision: FP32
[12/03/2020-15:26:48] [I] Calibration: 
[12/03/2020-15:26:48] [I] Refit: Disabled
[12/03/2020-15:26:48] [I] Safe mode: Disabled
[12/03/2020-15:26:48] [I] Save engine: new-tmp.engine
[12/03/2020-15:26:48] [I] Load engine: 
[12/03/2020-15:26:48] [I] Builder Cache: Enabled
[12/03/2020-15:26:48] [I] NVTX verbosity: 0
[12/03/2020-15:26:48] [I] Tactic sources: Using default tactic sources
[12/03/2020-15:26:48] [I] Input(s)s format: fp32:CHW
[12/03/2020-15:26:48] [I] Output(s)s format: fp32:CHW
[12/03/2020-15:26:48] [I] Input build shapes: model
[12/03/2020-15:26:48] [I] Input calibration shapes: model
[12/03/2020-15:26:48] [I] === System Options ===
[12/03/2020-15:26:48] [I] Device: 0
[12/03/2020-15:26:48] [I] DLACore: 
[12/03/2020-15:26:48] [I] Plugins:
[12/03/2020-15:26:48] [I] === Inference Options ===
[12/03/2020-15:26:48] [I] Batch: Explicit
[12/03/2020-15:26:48] [I] Input inference shapes: model
[12/03/2020-15:26:48] [I] Iterations: 10
[12/03/2020-15:26:48] [I] Duration: 3s (+ 200ms warm up)
[12/03/2020-15:26:48] [I] Sleep time: 0ms
[12/03/2020-15:26:48] [I] Streams: 1
[12/03/2020-15:26:48] [I] ExposeDMA: Disabled
[12/03/2020-15:26:48] [I] Data transfers: Enabled
[12/03/2020-15:26:48] [I] Spin-wait: Disabled
[12/03/2020-15:26:48] [I] Multithreading: Disabled
[12/03/2020-15:26:48] [I] CUDA Graph: Disabled
[12/03/2020-15:26:48] [I] Separate profiling: Disabled
[12/03/2020-15:26:48] [I] Skip inference: Disabled
[12/03/2020-15:26:48] [I] Inputs:
[12/03/2020-15:26:48] [I] === Reporting Options ===
[12/03/2020-15:26:48] [I] Verbose: Disabled
[12/03/2020-15:26:48] [I] Averages: 10 inferences
[12/03/2020-15:26:48] [I] Percentile: 99
[12/03/2020-15:26:48] [I] Dump refittable layers:Disabled
[12/03/2020-15:26:48] [I] Dump output: Disabled
[12/03/2020-15:26:48] [I] Profile: Disabled
[12/03/2020-15:26:48] [I] Export timing to JSON file: 
[12/03/2020-15:26:48] [I] Export output to JSON file: 
[12/03/2020-15:26:48] [I] Export profile to JSON file: 
[12/03/2020-15:26:48] [I] 
[12/03/2020-15:26:48] [I] === Device Information ===
[12/03/2020-15:26:48] [I] Selected Device: GeForce GTX 1660 Ti
[12/03/2020-15:26:48] [I] Compute Capability: 7.5
[12/03/2020-15:26:48] [I] SMs: 24
[12/03/2020-15:26:48] [I] Compute Clock Rate: 1.59 GHz
[12/03/2020-15:26:48] [I] Device Global Memory: 5944 MiB
[12/03/2020-15:26:48] [I] Shared Memory per SM: 64 KiB
[12/03/2020-15:26:48] [I] Memory Bus Width: 192 bits (ECC disabled)
[12/03/2020-15:26:48] [I] Memory Clock Rate: 6.001 GHz
[12/03/2020-15:26:48] [I] 
----------------------------------------------------------------
Input filename:   new-tmp.onnx
ONNX IR version:  0.0.6
Opset version:    11
Producer name:    pytorch
Producer version: 1.7
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
[12/03/2020-15:26:49] [W] [TRT] onnx2trt_utils.cpp:220: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
ERROR: builtin_op_importers.cpp:1601 In function importIf:
[8] Assertion failed: cond.is_weights() && cond.weights().count() == 1 && "If condition must be a initializer!"
[12/03/2020-15:26:49] [E] Failed to parse onnx file
[12/03/2020-15:26:49] [E] Parsing model failed
[12/03/2020-15:26:49] [E] Engine creation failed
[12/03/2020-15:26:49] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec # ./trtexec --onnx=tmp.onnx --saveEngine=tmp.engine
```

**Comment by wwdok** [comment]
solved by downgrade torch version from 1.7 to 1.5 ( 1.6 may works too)

**Comment by romil611** [comment]
I'm facing the same issue. I downloaded the pre-trained model from [Tensorflow model zoo](http://download.tensorflow.org/models/object_detection/tf2/20200711/efficientdet_d4_coco17_tpu-32.tar.gz). After that I tried running tf2onnx command:
`python3 -m tf2onnx.convert --saved-model models --output tf_model_op11.onnx --opset 11`
This gave an onnx file with int8 type. So I used Onnx graph surgeon tool to change the type:
` graph = gs.import_onnx(onnx.load("tf_model_op11.onnx"))`
`for inp in graph.inputs:`
`inp.dtype = np.float32`

After getting the onnx file, I ran the command:
`trtexec --onnx=model_op11.onnx --fp16=enable --workspace=5500 --batch=1 --saveEngine=model_op11.trt --verbose`

This gave me the same error as above. torch wasn't involved.

[06/11/2021-05:02:47] [I] === Model Options ===
[06/11/2021-05:02:47] [I] Format: ONNX
[06/11/2021-05:02:47] [I] Model: model_op11.onnx
[06/11/2021-05:02:47] [I] Output:
[06/11/2021-05:02:47] [I] === Build Options ===
[06/11/2021-05:02:47] [I] Max batch: 1
[06/11/2021-05:02:47] [I] Workspace: 5500 MB
[06/11/2021-05:02:47] [I] minTiming: 1
[06/11/2021-05:02:47] [I] avgTiming: 8
[06/11/2021-05:02:47] [I] Precision: FP32+FP16
[06/11/2021-05:02:47] [I] Calibration: 
[06/11/2021-05:02:47] [I] Safe mode: Disabled
[06/11/2021-05:02:47] [I] Save engine: model_op11.trt
[06/11/2021-05:02:47] [I] Load engine: 
[06/11/2021-05:02:47] [I] Builder Cache: Enabled
[06/11/2021-05:02:47] [I] NVTX verbosity: 0
[06/11/2021-05:02:47] [I] Inputs format: fp32:CHW
[06/11/2021-05:02:47] [I] Outputs format: fp32:CHW
[06/11/2021-05:02:47] [I] Input build shapes: model
[06/11/2021-05:02:47] [I] Input calibration shapes: model
[06/11/2021-05:02:47] [I] === System Options ===
[06/11/2021-05:02:47] [I] Device: 0
[06/11/2021-05:02:47] [I] DLACore: 
[06/11/2021-05:02:47] [I] Plugins:
[06/11/2021-05:02:47] [I] === Inference Options ===
[06/11/2021-05:02:47] [I] Batch: 1
[06/11/2021-05:02:47] [I] Input inference shapes: model
[06/11/2021-05:02:47] [I] Iterations: 10
[06/11/2021-05:02:47] [I] Duration: 3s (+ 200ms warm up)
[06/11/2021-05:02:47] [I] Sleep time: 0ms
[06/11/2021-05:02:47] [I] Streams: 1
[06/11/2021-05:02:47] [I] ExposeDMA: Disabled
[06/11/2021-05:02:47] [I] Spin-wait: Disabled
[06/11/2021-05:02:47] [I] Multithreading: Disabled
[06/11/2021-05:02:47] [I] CUDA Graph: Disabled
[06/11/2021-05:02:47] [I] Skip inference: Disabled
[06/11/2021-05:02:47] [I] Inputs:
[06/11/2021-05:02:47] [I] === Reporting Options ===
[06/11/2021-05:02:47] [I] Verbose: Enabled
[06/11/2021-05:02:47] [I] Averages: 10 inferences
[06/11/2021-05:02:47] [I] Percentile: 99
[06/11/2021-05:02:47] [I] Dump output: Disabled
[06/11/2021-05:02:47] [I] Profile: Disabled
[06/11/2021-05:02:47] [I] Export timing to JSON file: 
[06/11/2021-05:02:47] [I] Export output to JSON file: 
[06/11/2021-05:02:47] [I] Export profile to JSON file: 

Input filename:   model_op11.onnx
ONNX IR version:  0.0.7
Opset version:    11
Producer name:    
Producer version: 
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
[06/11/2021-05:02:49] [V] [TRT] ModelImporter.cpp:103: Parsing node: StatefulPartitionedCall/Preprocessor/ResizeToRange/cond [If]
[06/11/2021-05:02:49] [V] [TRT] ModelImporter.cpp:119: Searching for input: StatefulPartitionedCall/Preprocessor/ResizeToRange/Less:0
[06/11/2021-05:02:49] [V] [TRT] ModelImporter.cpp:125: StatefulPartitionedCall/Preprocessor/ResizeToRange/cond [If] inputs: [StatefulPartitionedCall/Preprocessor/ResizeToRange/Less:0 -> ()], 
ERROR: builtin_op_importers.cpp:1554 In function importIf:
[8] Assertion failed: cond.is_weights() && cond.weights().count() == 1 && "If condition must be a initializer!"
[06/11/2021-05:02:49] [E] Failed to parse onnx file
[06/11/2021-05:02:49] [E] Parsing model failed
[06/11/2021-05:02:49] [E] Engine creation failed
[06/11/2021-05:02:49] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec # trtexec --onnx=model_op11.onnx --fp16=enable --workspace=5500 --batch=1 --saveEngine=model_op11.trt --verbose

[The whole verbose output](https://drive.google.com/file/d/1OJcaBlWlndffw1Q_P4aCuEqxd5ZeN824/view?usp=sharing)
